algorithms/beefy/src/lib.rs

```diff
@@ -204,10 +205,15 @@ where
 		}
 
 		// Verify timestamp extrinsic
-		let proof = &*parachain_header.extrinsic_proof;
-		let ext = &*parachain_header.timestamp_extrinsic;
-		let extrinsic_root = decoded_para_header.extrinsics_root;
-		<H as HostFunctions>::verify_timestamp_extrinsic(extrinsic_root, proof, ext)?;
+		// Timestamp extrinsic should be the first inherent and hence the first extrinsic
+		// https://github.com/paritytech/substrate/blob/d602397a0bbb24b5d627795b797259a44a5e29e9/primitives/trie/src/lib.rs#L99-L101
+		let timestamp_ext_key = codec::Compact(0u32).encode();
+		sp_trie::verify_trie_proof::<LayoutV0<H::BlakeTwo256>, _, _, _>(
```

Comment:
Shouldn't we use the host function, `sp_io::trie::blake2_256_verify_proof` here ?

Reply:
Yeah that wouldn't be chain agnostic, remember we have to support near and cosmos (via ics-8 wasm client)

Reply:
Okay, makes sense.

Reply:
The `H::BlakeTwo256: hash_db::Hashing` so we can implement it using the different host functions for the different runtimes